source/opt/set_spec_constant_default_value_pass.cpp

```diff
@@ -248,5 +250,85 @@ bool SetSpecConstantDefaultValuePass::Process(ir::Module* module) {
   return modified;
 }
 
+// A helper function for parsing spec constant default value string. Returns
+// true if the char pointed by the given pointer is considered as the end of
+// the string, which means it is a '\0' or '}'.
+inline bool IsEnd(const char* const ptr) {
+  if (!ptr) return false;
+  return *ptr == '\0' || *ptr == '}';
+}
+
+// A helper function for parsing spec constant default value string. Returns
+// true if the char pointed by the given pointer is considered as the content
+// of an entry, which means it is not a space, end, ':' or ','.
+inline bool IsEntryContent(const char* const ptr) {
+  if (!ptr) return false;
+  return std::isspace(*ptr) == 0 && !IsEnd(ptr) && *ptr != ',' && *ptr != ':';
```

Comment:
Everything beyond the isspace is a good candidate for
  if ( std::strchr("},:", *ptr) == nullptr )  {  I am entry content }
(I'm even half tempted to replace the isspace with it too, except for locale complications...  Meh.)